src/components/CollectionDetailPage.vue

```diff
@@ -415,16 +419,21 @@ export default {
       if (page !== undefined) {
         this.$store.commit('collections/UPDATE_PAGINATION_CURRENT_PAGE', parseInt(page, 10));
       }
-      this.$store.dispatch('collections/LOAD_COLLECTION', this.collection).then((res) => {
-        this.collection = res;
-        this.fetching = false;
-      });
+      if (this.collection.uid === 'co_all') {
```

Comment:
It is better to check for null/undefined `this.collection.uid == null` instead of a magic collection Id. Usually magic numbers produce bugs in long run (for example if later someone wants to check if a real collection has been loaded by checking the UID for null/undefined).